addons/binding/org.openhab.binding.co2signal/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="co2signal" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/thing-description/v1.0.0 http://eclipse.org/smarthome/schemas/thing-description-1.0.0.xsd">
+
+	<!-- CO2 Signal Thing -->
+	<thing-type id="co2signal">
+		<label>CO2 Signal</label>
+		<description>
+		Provides information about the CO2 intensity of produced electricity of a given location.
+		In order to receive the data, you must register an account on https://www.co2signal.com/ and get your API token.
+        </description>
+
+		<channels>
+            <channel id="countryCode" typeId="countryCode" />
+			<channel id="carbonIntensity" typeId="carbonIntensity" />
+			<channel id="fossilFuelPercentage" typeId="fossilFuelPercentage" />
+		</channels>
+
+		<config-description>
+			<parameter name="apikey" type="text" required="true">
+				<context>password</context>
+				<label>API Key</label>
+				<description>Data-platform token to access the www.co2signal.com service</description>
+			</parameter>
+			<parameter name="location" type="text" required="false"
```

Comment:
I would assume we have a type for coordinates, and if you are interested in the current location of the user that should even be available directly without the need for configuration.

Reply:
Can't find anything about this

Reply:
@Jsurf This PR might be helpful: https://github.com/openhab/openhab2-addons/pull/2692/files

Reply:
Great! Will have a look

Reply:
Implemented Auto Discovery